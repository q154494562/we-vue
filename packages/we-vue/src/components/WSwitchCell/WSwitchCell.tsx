// Utils
import mixins from '@/utils/mixins'

import WCell from '../WCell'
import WSwitch from '../WSwitch'

// Mixins
import Toggleable from '@/mixins/toggleable'

export default mixins(
  Toggleable
).extend({
  name: 'w-switch-cell',

  components: {
    WCell,
    WSwitch,
  },

  props: {
    title: String,
    disabled: Boolean,
  },

  render () {
    return (
      <WCell title={this.title}>
        <WSwitch
          slot="ft"
          vModel={this.isActive}
          disabled={this.disabled}
        />
      </WCell>
    )
  },
})
