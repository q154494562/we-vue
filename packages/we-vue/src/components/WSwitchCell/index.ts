import WSwitchCell from './WSwitchCell'

export { WSwitchCell }
export default WSwitchCell
