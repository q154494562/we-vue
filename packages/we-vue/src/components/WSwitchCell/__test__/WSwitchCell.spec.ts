import { mount } from '@vue/test-utils'
import WSwitchCell from '../'

describe('switch-cell', () => {
  test('create', () => {
    const wrapper = mount(WSwitchCell, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-switch-cell')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('click', () => {
    let wrapper = mount(WSwitchCell, {
      propsData: {
        value: true,
      },
    })

    // first click
    wrapper.find('.wv-switch').trigger('click')
    expect(wrapper.vm.isActive).toBe(false)

    // second click
    wrapper.find('.wv-switch').trigger('click')
    expect(wrapper.vm.isActive).toBe(true)

    // click a disabled switch
    wrapper = mount(WSwitchCell, {
      propsData: {
        disabled: true,
        value: true,
      },
    })

    wrapper.find('.wv-switch').trigger('click')

    expect(wrapper.vm.isActive).toBe(true)
  })

  test('watch value', () => {
    const wrapper = mount(WSwitchCell, {
      propsData: {
        value: false,
      },
    })

    wrapper.setProps({
      value: false,
    })

    expect(wrapper.vm.isActive).toBe(false)

    wrapper.setProps({
      value: true,
    })

    expect(wrapper.vm.isActive).toBe(true)
  })
})
