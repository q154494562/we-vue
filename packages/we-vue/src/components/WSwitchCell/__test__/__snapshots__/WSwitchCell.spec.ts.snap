// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switch-cell create 1`] = `
<div class="weui-cell">
  <div class="weui-cell_hd"></div>
  <div class="weui-cell__bd">
    <p></p>
  </div>
  <div class="weui-cell__ft">
    <div class="wv-switch">
      <div class="background"></div>
      <div class="thumb" style="transform: translate3d(0px, 0, 0);"></div>
    </div>
  </div>
</div>
`;
